.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ "main", "testnet" ]
+  pull_request:
+    branches: [ "main", "testnet" ]
```

Comment:
[Recommendation] Having a localnet or devnet stage here would be good. Show it works locally and mess around first. Testnet is a good testing ground for functionally correct and working updates. Having a dedicated devnet environment would be a good idea

Reply:
- Good idea, this feels like a better flow for integration tests than going straight to testnet. 
- Also agree that a dedicated env is definitely the way to go, spinning it up during the CI workflow would make the test process insanely long.
  -  If it still ends up being a longer running test, we can update triggers to only run it for specific pre-release tags or something instead of on every push/PR so it doesn't slow down the dev cycle